tests/js/articles.spec.js

```diff
@@ -0,0 +1,178 @@
+
+    // Add richtext inputs.
+    await clearPreviousTextWithPlaceholder( 'Enter title' );
+    await typeInInputWithPlaceholderLabel( 'Enter title', 'News' );
+    await typeInInputWithPlaceholderLabel( 'Enter description', 'Latest news details' );
+
+    // Add loadmore button text in wysiwyg editor.
+    await clearPreviousTextWithLabel( 'Button Text' );
+    await typeInInputWithPlaceholderLabel( 'Enter text', 'More News' );
+
+    // Add Setting inputs
+    await openSidebarPanelWithTitle( 'Setting' );
+    await clearPreviousTextWithLabel( 'Button Text' );
+    await typeInInputWithLabel( 'Button Text', 'Read more' );
+    await typeInInputWithLabel( 'Button Link', 'https://www.greenpeace.org' );
+    await clickElementByText( 'label', 'Open in a new Tab' );
+    await clearPreviousTextWithLabel( 'Articles count' );
+    await typeInInputWithLabel( 'Articles count', '4' );
+    await typeInDropdownWithLabel( 'Select Tags', 'Climate' );
+    await typeInDropdownWithLabel( 'Post Types', 'Story' );
+    await clickElementByText( 'label', 'Ignore categories' );
+
+    expect( await getEditedPostContent() ).toMatchSnapshot();
+  }, 50000 );
+
+  it ( 'should show a warning if the URL is wrong', async () => {
+    await typeInInputWithLabel( 'Button Link', 'this is not a URL' );
+
+    // The warning component should appear.
+    await expect( page ).toMatchElement( '.edit-post-sidebar .input_error' );
+
+    // The warning text should appear.
+    await expect( page ).toMatch( ARTICLES_INVALID_URL_WARNING );
+  } );
+
+  it ( 'test Manual override', async () => {
+    await openSidebarPanelWithTitle( 'Setting' );
+
+    await typeInDropdownWithLabel( 'Manual override', 'Duis posuere' );
+    await typeInDropdownWithLabel( 'Manual override', 'Mauris quis dictum magna' );
+    await typeInDropdownWithLabel( 'Manual override', 'Lorem ipsum dolor sit amet' );
+
+    // The only 3 manually override post should appear in articles block.
+    await expect(page).toMatchElement('article:nth-child(1) h4 a', { text: 'Duis posuere' });
+    await expect(page).toMatchElement('article:nth-child(2) h4 a', { text: 'Mauris quis dictum magna' });
+    await expect(page).toMatchElement('article:nth-child(3) h4 a', { text: 'Lorem ipsum dolor sit amet' });
+
+    // The "Load more" should not appear.
+    await expect( page ).not.toMatchElement( '.block-editor .article-load-more' );
+
+    expect( await getEditedPostContent() ).toMatchSnapshot();
+  } , 50000 );
+
+  it ( 'test Load more button', async ( done ) => {
+    await selectBlockByName( 'planet4-blocks/articles' );
+
+    // Add Setting inputs
+    await openSidebarPanelWithTitle( 'Setting' );
+    await clearPreviousTextWithLabel( 'Articles count' );
+    await typeInInputWithLabel( 'Articles count', '4' );
+
+    // Publish the page.
+    await publishPost();
+
+    // Test articles block on frontend.
+    await clickElementByText( 'a', 'View Page' );
```

Comment:
I think navigating away from the editor and to the frontend page this way will make the tests slow if we do that a lot. But we actually don't need to enter the articles count "manually" during this test. We already have a test which confirms that entering into the editor will put the value in the attribute. So in this test we could just have a snapshot with a fixed page (and fixed other content) where we could navigate directly to, and perform the assertions below this line.

Reply:
Maybe it would make sense to put the frontend and editor tests into separate files? Then the editor test would correspond be for https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/master/assets/src/blocks/Articles/ArticlesEditor.js and the rest would be for https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/master/assets/src/blocks/Articles/ArticlesFrontend.js

Reply:
Yes, I agree with you. Created a separate [tests](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/354/commits/44fffdf3ad5fd3c6bb11bff4f2c1754787a7c544#diff-681f3eaecf153bfb54d00857cbae7c41) for frontend.